### ISSUE TITLE
"fieldProps" exposed from "onChange" method handler are one step behind

### ISSUE THREAD

**Issue by kettanaito** [kettanaito/react-advanced-form#171]
## What
It seems that `fieldProps` Object of a custom field inside `onChange` handler is one step behind the actual state of the field. The issue may be relative to `fields` Object as well.

## Why
This issue is caused by the sequence of method within `Form.handleFieldChange`:
```js
const { nextFieldProps } = await this.updateField();

appropriateValidation();

onChange({ fieldProps: nextFieldProps });
```
What is required is for `onChange` callback to accepts the props *after* the `appropriateValidation()` finished.

**Comment by kettanaito** [comment]
It is troublesome right now to await the debounced validation, as that is a sync function which doesn't return the same payload as `Form.validateField()`.

**Comment by kettanaito** [comment]
The changelog is listed in the pull request to this issue.

Merged and published under [`0.19.4`](https://github.com/kettanaito/react-advanced-form/commit/1c0f5aea0ee9c7645f25e124114e09b49a6d17d1). Closing.
